examples/main.tf

```diff
@@ -110,3 +110,7 @@ resource "powerstore_protectionpolicy" "terraform-provider-test" {
 data "powerstore_volume" "test1" {
         name = "tf_vol"
 }
+
+data "powerstore_snapshotrule" "test1" {
+       name = "tf_snapshotRule"
+}
```

Comment:
My suggestion is that better we maintain the hierarchy as this : examples/data-sources/powerstore_protectionpolicy/protection_policy.tf . You can refer here https://github.com/dell/terraform-provider-powerstore/pull/44/files . Also please provide proper comments to make it more readable

Reply:
added the same in a separate file, this is the common file